core/block_rendering_rewrite/measurables.js

```diff
@@ -208,6 +263,86 @@ Blockly.BlockRendering.ExternalValueInput = function(input) {
 goog.inherits(Blockly.BlockRendering.ExternalValueInput,
     Blockly.BlockRendering.Input);
 
+/**
+ * An object containing information about the space a previous connection takes
+ * up during rendering.
+ * @package
+ * @constructor
+ */
+Blockly.BlockRendering.PreviousConnection = function() {
+  Blockly.BlockRendering.PreviousConnection.superClass_.constructor.call(this);
+  this.renderRect = null;
+  this.type = 'previous connection';
+  this.height = BRC.MEDIUM_PADDING;
+  this.width = BRC.NOTCH_WIDTH;
+
+};
+goog.inherits(Blockly.BlockRendering.PreviousConnection, Blockly.BlockRendering.Measurable);
+
+/**
+ * An object containing information about the space a next connection takes
+ * up during rendering.
+ * @package
+ * @constructor
+ */
+Blockly.BlockRendering.NextConnection = function() {
+  Blockly.BlockRendering.NextConnection.superClass_.constructor.call(this);
+  this.renderRect = null;
+  this.type = 'next connection';
+  this.height = BRC.MEDIUM_PADDING;
+  this.width = BRC.NOTCH_WIDTH;
+};
+goog.inherits(Blockly.BlockRendering.NextConnection, Blockly.BlockRendering.Measurable);
+
+/**
+ * An object containing information about the space a hat takes up during
+ * rendering.
+ * @package
+ * @constructor
+ */
+Blockly.BlockRendering.Hat = function() {
+  Blockly.BlockRendering.Hat.superClass_.constructor.call(this);
+  this.renderRect = null;
+  this.type = 'hat';
+  this.height = BRC.MEDIUM_PADDING;
```

Comment:
Add a separate constant for this height as well.

Reply:
...though actually, this isn't the height up, right?  It's the height down into the row?  If so, does this need to have any height at all, or can we say its height is 0?